### ISSUE TITLE
`unsubscriptable-object` when indexing a numpy array inside a list.

### ISSUE THREAD

**Issue by acxz** [PyCQA/pylint#3956]
### Steps to reproduce
The following script gives me the `unsubscriptable-object` warning

```
"""."""
import numpy as np

x_vec_list = [None] * 2
x_vec_list[0] = np.zeros(2)
x_vec_list[1] = np.zeros(2)

x1_vec = [x_vec[0] for x_vec in x_vec_list]

print(x1_vec)
```

### Current behavior
```
************* Module test
test.py:8:10: E1136: Value 'x_vec' is unsubscriptable (unsubscriptable-object)

-----------------------------------
Your code has been rated at 1.67/10
```

### Expected behavior
No warning

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.6 (default, Sep 30 2020, 04:00:38) 
[GCC 10.2.0]
```

First incorrectly reported at #3139

**Comment by hippo91** [comment]
@acxz thanks for your report.
The problem here is probably due to the fact that `pylint` doesn't analyze the control flow. In the snippet you gave, the list `x_vec_list` is firstly a list of None and then a list of numpy arrays. 
For `pylint` (and `astroid` that is the `pylint`'s inference engine) `x_vec_list` is a list of None and doesn't take into account second assignment, that's why it raises this message.
For example, the following code is almost identical to yours but doesn't trigger the `unsubscriptable-object` message emission :
```
import numpy as np

x_vec_list = [np.zeros(2), np.zeros(2)]

x1_vec = [x_vec[0] for x_vec in x_vec_list]

print(x1_vec)
```

I close this issue because in the way `pylint` and `astroid` work, it doesn't seems to be a bug.

**Comment by acxz** [comment]
@hippo91 that makes sense, thanks for looking into this!
